### DISCUSSION TITLE
Export LUTs created by the Color Grading Filter

### DISCUSSION THREAD

**Discussion by zaytri** [Xaymar/obs-stream-effects#853] [Ideas & Feedback]
### Explain the Feature/Enhancement

It would be great to export LUTs so that we can use the same ones in other applications, or even import LUTs to fine tune them using the Color Grading filter!

### What makes this feature/enhancement useful for StreamFX and its users?

Provides a method to easily share Color Grading settings!

### Checklist

- [ ] I am offering a monetary bounty/reward for the implementation of this Feature/Enhancement.
- [X] There is no other way to currently do this in StreamFX or OBS Studio.

**Comment by Xaymar** [comment]
- Importing LUTs is already available as the LUT filter, simply add it and a Color Grading filter to it and you have your fine tuning capability.
- Exporting LUTs is planned eventually.

**Comment by Xaymar** [comment]
Here's a few potential LUT file formats to consider:

- .png, .bmp, .webp: Mostly matches internal structure of generated LUTs, not much to do here.
- .cube: 
    - Adobe: https://wwwimages2.adobe.com/content/dam/acom/en/products/speedgrade/cc/pdfs/cube-lut-specification-1.0.pdf
    - BMD: https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=40284
- .ctf: http://docs.autodesk.com/smoke2013/index.html?url=files/GUID-460BA05D-3AB7-4BE2-AD30-01F9D3440CD8.htm,topicNumber=d30e106812
- .3dl: https://community.foundry.com/discuss/topic/103636/format-spec-for-3dl?mode=Post&postID=895260

Out of these, .png and .cube seem the easiest to support.

**Comment by Xaymar** [comment]
Rejected the *Import LUT* request based on these facts:

- It's already possible to import PNG and CUBE LUT files into OBS Studio via the Apply LUT filter, which will produce identical results as Color Grading would.
- It's not feasible to reverse engineer the parameters that generated the imported LUT file, as there are `2 ^ 32 ^ (5+3+5+4+9+4+4+4+4)` possible solutions. This would take an eternity to figure out.
